drivers/volume/csi/csi.go

```diff
@@ -1234,10 +1234,12 @@ func (c *csi) downloadObject(
 	if err != nil {
 		return nil, err
 	}
-	if restoreLocation.Location.EncryptionKey != "" {
-		if data, err = crypto.Decrypt(data, restoreLocation.Location.EncryptionKey); err != nil {
-			return nil, err
+	if restoreLocation.Location.EncryptionV2Key != "" {
+		var decryptData []byte
+		if decryptData, err = crypto.Decrypt(data, restoreLocation.Location.EncryptionV2Key); err != nil {
+			return data, nil
```

Comment:
nit: add a log line indicating why decrypt failed and us returning the data as is

Reply:
fixed